From af108a6787b01535519f8724be60c3a95e5b889f Mon Sep 17 00:00:00 2001
From: sub77 <sub77@ymail.com>
Date: Mon, 30 Oct 2017 14:34:18 +0000
Subject: [PATCH 2/2] Revert "Camera: CameraHardwareInterface changes to
 support Extended FD"

This reverts commit 5b6fa9f9ff6e3c2e861b16b59ea8af96a2706922.
---
 .../libcameraservice/device1/CameraHardwareInterface.cpp   | 14 --------------
 .../libcameraservice/device1/CameraHardwareInterface.h     |  7 +------
 2 files changed, 1 insertion(+), 20 deletions(-)

diff --git a/services/camera/libcameraservice/device1/CameraHardwareInterface.cpp b/services/camera/libcameraservice/device1/CameraHardwareInterface.cpp
index 9a1ec29..469c86c 100644
--- a/services/camera/libcameraservice/device1/CameraHardwareInterface.cpp
+++ b/services/camera/libcameraservice/device1/CameraHardwareInterface.cpp
@@ -137,20 +137,6 @@ hardware::Return<void> CameraHardwareInterface::dataCallback(
     return hardware::Void();
 }
 
-hardware::Return<void> CameraHardwareInterface::QDataCallback(
-        DataCallbackMsg msgType, uint32_t data, uint32_t bufferIndex,
-        const vendor::qti::hardware::camera::device::V1_0::QCameraFrameMetadata& metadata) {
-    if (mHidlMemPoolMap.count(data) == 0) {
-        ALOGE("%s: memory pool ID %d not found", __FUNCTION__, data);
-        return hardware::Void();
-    }
-    camera_frame_metadata_t md;
-    md.number_of_faces = metadata.faces.size();
-    md.faces = (camera_face_t*) metadata.faces.data();
-    sDataCb((int32_t) msgType, mHidlMemPoolMap.at(data), bufferIndex, &md, this);
-    return hardware::Void();
-}
-
 hardware::Return<void> CameraHardwareInterface::dataCallbackTimestamp(
         DataCallbackMsg msgType, uint32_t data,
         uint32_t bufferIndex, int64_t timestamp) {
diff --git a/services/camera/libcameraservice/device1/CameraHardwareInterface.h b/services/camera/libcameraservice/device1/CameraHardwareInterface.h
index 42d88b8..1c38d00 100644
--- a/services/camera/libcameraservice/device1/CameraHardwareInterface.h
+++ b/services/camera/libcameraservice/device1/CameraHardwareInterface.h
@@ -29,7 +29,6 @@
 #include <hardware/camera.h>
 
 #include <common/CameraProviderManager.h>
-#include <vendor/qti/hardware/camera/device/1.0/IQCameraDeviceCallback.h>
 
 namespace android {
 
@@ -86,7 +85,7 @@ typedef void (*data_callback_timestamp_batch)(
 
 class CameraHardwareInterface :
         public virtual RefBase,
-        public virtual vendor::qti::hardware::camera::device::V1_0::IQCameraDeviceCallback,
+        public virtual hardware::camera::device::V1_0::ICameraDeviceCallback,
         public virtual hardware::camera::device::V1_0::ICameraDevicePreviewCallback {
 
 public:
@@ -433,10 +432,6 @@ private:
             hardware::camera::device::V1_0::DataCallbackMsg msgType,
             const hardware::hidl_vec<
                     hardware::camera::device::V1_0::HandleTimestampMessage>&) override;
-    hardware::Return<void> QDataCallback(
-            hardware::camera::device::V1_0::DataCallbackMsg msgType,
-            uint32_t data, uint32_t bufferIndex,
-            const vendor::qti::hardware::camera::device::V1_0::QCameraFrameMetadata& metadata) override;
 
     /**
      * Implementation of android::hardware::camera::device::V1_0::ICameraDevicePreviewCallback
-- 
2.1.2

